wordpress-mobile/WordPress-Android
# Feature/8501 sc verticals ui

Fixes #8501 

Note:
1) This PR contains changes from #8618 - for easier review, merge #8618 first.
2)This is not a final version of the screen but just a first iteration. 

Adds a View layer for the SiteCreation Business Verticals screen.


To test:
1. Open the SiteCreation flow
2. Verify that the following requirements are met (rotate your device)

Header + Skip button
- visible only when the edittext is empty -> when the user enters the screen or whenever when the user deletes the content of the edittext

EditText
- typeahead/autocompletion functionality - with throttle
- contains icon on the left whenever the app is fetching suggestions the icon is replaced with an infinite progressbar
- contains clear button on the right - visible only when the edittext is not empty
